src/lidar/read_rp_lidar.cpp

```diff
@@ -61,16 +50,15 @@ void readLidarLoop() {
 /**
  * @brief Startsup RPLidar with default settings
  */
-bool initializeLidar() {
+bool initializeLidar(double max_dist = 16) {
     if (!lidar_initialized) {
         if (!rp_lidar.checkHealth()) {
             perror("failed to connect to rp lidar");
         } else {
-            rp_lidar.setMaxDistance(16.0);
-            points_lock.lock();
+            rp_lidar.setMaxDistance(max_dist);
+            std::lock_guard<std::mutex> lk(points_lock);
             lidar_thread = std::thread(&readLidarLoop);
             lidar_initialized = true;
-            points_lock.unlock();
         }
     }
     return lidar_initialized;
```

Comment:
`lidar_initialized` should be protected by the mutex, so create the `lock_guard` just before the `if(!lidar_initialized)` line. This is necessary to ensure that the initialization state is synchronized correctly.